source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -1,79 +1,717 @@
+                                              ILog log,
+                                              ScriptSyntax scriptSyntax,
+                                              ICommandLineRunner commandLineRunner,
+                                              Dictionary<string, string> environmentVars,
+                                              string workingDirectory)
             {
-                case ScriptSyntax.Bash:
-                    contextFile = "KubectlBashContext.sh";
-                    break;
-                case ScriptSyntax.PowerShell:
-                    contextFile = "KubectlPowershellContext.ps1";
-                    break;
-                default:
-                    throw new InvalidOperationException($"No kubernetes context wrapper exists for {syntax}");
+                this.variables = variables;
+                this.log = log;
+                this.scriptSyntax = scriptSyntax;
+                this.commandLineRunner = commandLineRunner;
+                this.environmentVars = environmentVars;
+                this.workingDirectory = workingDirectory;
             }
 
-            var k8sContextScriptFile = Path.Combine(workingDirectory, $"Octopus.{contextFile}");
-            var contextScript = embeddedResources.GetEmbeddedResourceText(Assembly.GetExecutingAssembly(), $"Calamari.Kubernetes.Scripts.{contextFile}");
-            fileSystem.OverwriteFile(k8sContextScriptFile, contextScript);
-            return k8sContextScriptFile;
+            public CommandResult Execute(string accountType)
+            {
+                var errorResult = new CommandResult(string.Empty, 1);
+
+                foreach (var proxyVariable in ProxyEnvironmentVariablesGenerator.GenerateProxyEnvironmentVariables())
+                {
+                    environmentVars[proxyVariable.Key] = proxyVariable.Value;
+                }
+
+                var kubeConfig = ConfigureCliExecution();
+
+                if (!TrySetKubectl())
+                {
+                    return errorResult;
+                }
+
+                var @namespace = variables.Get(SpecialVariables.Namespace);
+                if (string.IsNullOrEmpty(@namespace))
+                {
+                    log.Verbose("No namespace provided. Using default");
+                    @namespace = "default";
+                }
+
+                if (!TrySetupContext(kubeConfig, @namespace, accountType))
+                {
+                    return errorResult;
+                }
+
+                CreateNamespace(@namespace);
+
+                var outputKubeConfig = variables.GetFlag(SpecialVariables.OutputKubeConfig);
+                if (outputKubeConfig)
+                {
+                    ExecuteKubectlCommand("config", "view");
+                }
+
+                return new CommandResult(string.Empty, 0);
+            }
+
+            bool TrySetupContext(string kubeConfig, string @namespace, string accountType)
```

Comment:
Is there a way we could break this method down a bit further? It is 300+ lines of nested statements, and is pretty tough to get your head around.